code/go/0chain.net/core/transaction/http.go

```diff
@@ -173,72 +173,106 @@ func VerifyTransaction(txnHash string, chain *chain.Chain) (*Transaction, error)
 }
 
 func MakeSCRestAPICall(scAddress string, relativePath string, params map[string]string, chain *chain.Chain, handler SCRestAPIHandler) ([]byte, error) {
+	var resMaxCounterBody []byte
+	resBodies := make(map[string][]byte)
+
+	var hashMaxCounter int
+	hashCounters := make(map[string]int)
+
 	network := zcncore.GetNetwork()
 	numSharders := len(network.Sharders)
 	sharders := util.GetRandom(network.Sharders, numSharders)
-	responses := make(map[string]int)
-	entityResult := make(map[string][]byte)
-	var retObj []byte
-	maxCount := 0
+
 	for _, sharder := range sharders {
-		urlString := fmt.Sprintf("%v/%v%v%v", sharder, SC_REST_API_URL, scAddress, relativePath)
-		urlObj, _ := url.Parse(urlString)
-		q := urlObj.Query()
-		for k, v := range params {
-			q.Add(k, v)
-		}
-		urlObj.RawQuery = q.Encode()
-		h := sha1.New()
-		var netTransport = &http.Transport{
+		// Make one or more requests (in case of unavailability, see 503/504 errors)
+		var err error
+		var res *http.Response
+		var counter int = 3 // TODO: better create a const
+
+		netTransport := &http.Transport{
 			Dial: (&net.Dialer{
-				Timeout: 5 * time.Second,
+				Timeout: 5 * time.Second, // TODO: better create a const
 			}).Dial,
-			TLSHandshakeTimeout: 5 * time.Second,
+			TLSHandshakeTimeout: 5 * time.Second, // TODO: better create a const
 		}
-		var netClient = &http.Client{
-			Timeout:   time.Second * 10,
+
+		netClient := &http.Client{
+			Timeout:   time.Second * 10, // TODO: create a const
 			Transport: netTransport,
 		}
-		response, err := netClient.Get(urlObj.String())
+
+		uString := fmt.Sprintf("%v/%v%v%v", sharder, SC_REST_API_URL, scAddress, relativePath)
+		u, _ := url.Parse(uString)
+		q := u.Query()
+		for k, v := range params {
+			q.Add(k, v)
+		}
+		u.RawQuery = q.Encode()
+
+		for counter > 0 {
+			res, err := netClient.Get(u.String())
+			if err != nil { break }
+
+			// TODO: better create an utility function as isAvailable or so
```

Comment:
If you think there should be a unitlity function, then create it. Does this block of code get used anywhere else? if so then it looks like a good idea.